cherab/tools/observers/spectroscopic.py

```diff
@@ -60,163 +516,375 @@ def sight_lines(self):
 
-    def plot_spectra(self, unit='J', ymax=None, extras=True):
+    def observed_spectrum(self, item=0):
         """
-        Plot the observed spectrum.
+        Returns observed spectrum for a given pipeline.
+        :param item: The index or name of the pipeline. Default: 0.
+
+        :rtype: Spectrum
+        """
+
+        pipeline = self.get_pipeline(item)
+        if isinstance(pipeline, PowerPipeline0D):
+            spectrum = Spectrum(self.min_wavelength, self.max_wavelength, 1)
+            spectrum.samples[0] = pipeline.value.mean / (self.max_wavelength - self.min_wavelength)
+        elif isinstance(pipeline, SpectralPowerPipeline0D):
+            if not pipeline.samples:
+                raise ValueError("No spectrum has been observed.")
+            spectrum = Spectrum(pipeline.min_wavelength, pipeline.max_wavelength, pipeline.bins)
+            spectrum.samples = pipeline.samples.mean
+        else:
+            raise TypeError("Method 'observed_spectrum' supports only PowerPipeline0D, SpectralPowerPipeline0D and their instances. "
+                            "Pipeline {} of type {} is not supported.".format(item, self.__class__.__name__))
+        return spectrum
+
+    def plot_spectra(self, item=0, unit='J', ymax=None, extras=True):
+        """
+        Plot the observed spectrum for a given pipeline.
+
+        :param item: The index or name of the pipeline. Default: 0.
+        :param str unit: Plots the spectrum in J/s (units='J'),
+                         or in ph/s (units='ph').
+        :param float ymax: Upper limit of y-axis.
+        :param bool extras: If True, set title, axis labels and ymax.
         """
 
         if unit == 'J':
-            # Spectrum objects are already in J/s/m2/str/nm
-            spectrum = self.observed_spectrum
+            # Spectrum objects are already in J/s
+            spectrum = self.observed_spectrum(item)
         elif unit == 'ph':
-            # turn the samples into ph/s/m2/str/nm
-            spectrum = self.observed_spectrum.new_spectrum()
-            spectrum.samples = self.observed_spectrum.to_photons()
+            # turn the samples into ph/s
+            spectrum_observed = self.observed_spectrum(item)
+            spectrum = spectrum_observed.new_spectrum()
+            spectrum.samples = spectrum_observed.to_photons()
         else:
             raise ValueError("unit must be 'J' or 'ph'.")
 
-        plt.plot(spectrum.wavelengths, spectrum.samples)
+        if spectrum.samples.size > 1:
+            plt.plot(spectrum.wavelengths, spectrum.samples, label=self.name)
+        else:
+            pipeline = self.get_pipeline(item)
+            if isinstance(pipeline, SpectralPowerPipeline0D):
+                plt.plot(spectrum.wavelengths, spectrum.samples, marker='o', ls='none', label=self.name)
+            elif isinstance(pipeline, PowerPipeline0D):
+                plt.plot(spectrum.wavelengths, spectrum.total(), marker='o', ls='none', label=self.name)
```

Comment:
To avoid confusion between `SpectralPowerPipeline0D` and `PowerPipeline0D` I would suggest adding only a legend record in case of `PowerPipeline0D`, something like:
plt.plot([], [], marker='none', ls='none', label=self.name + " {} {}".format(value, unit))